tikv/cop_handler_test.go

```diff
@@ -0,0 +1,363 @@
+		if err != nil {
+			return nil, err
+		}
+	}
+	store := NewMVCCStore(dbs, settings.dbPath, safePoint)
+	svr := &Server{
+		mvccStore: store,
+	}
+	return &TestStore{store, svr}, nil
+}
+
+func PrepareTestTableData(t *testing.T, keyNumber int, tableId int64) *data {
+	stmtCtx := new(stmtctx.StatementContext)
+	colIds := []int64{1, 2, 3}
+	colTypes := []*types.FieldType{
+		types.NewFieldType(mysql.TypeLonglong),
+		types.NewFieldType(mysql.TypeString),
+		types.NewFieldType(mysql.TypeDouble),
+	}
+	colInfos := make([]*tipb.ColumnInfo, 3)
+	colTypeMap := map[int64]*types.FieldType{}
+	for i := 0; i < 3; i++ {
+		colInfos[i] = &tipb.ColumnInfo{
+			ColumnId: colIds[i],
+			Tp:       int32(colTypes[i].Tp),
+		}
+		colTypeMap[colIds[i]] = colTypes[i]
+	}
+	rowValues := map[int64][]types.Datum{}
+	encodedKVDatas := make([]*encodedKVData, keyNumber)
+	for i := 0; i < keyNumber; i++ {
+		datum := types.MakeDatums(i, "abc", 10.0)
+		rowValues[int64(i)] = datum
+		rowEncodedData, err := tablecodec.EncodeRow(stmtCtx, datum,
+			colIds, nil, nil)
+		require.Nil(t, err)
+		rowKeyEncodedData := tablecodec.EncodeRowKeyWithHandle(tableId, int64(i))
+		encodedKVDatas[i] = &encodedKVData{encodedRowKey: rowKeyEncodedData, encodedRowValue: rowEncodedData}
+	}
+	return &data{
+		colInfos:       colInfos,
+		encodedKVDatas: encodedKVDatas,
+		rowValues:      rowValues,
+		colTypes:       colTypeMap,
+	}
+}
+
+func getTestPointRange(tableId int64, handle int64) kv.KeyRange {
+	startKey := tablecodec.EncodeRowKeyWithHandle(tableId, handle)
+	endKey := make([]byte, len(startKey))
+	copy(endKey, startKey)
+	convertToPrefixNext(endKey)
+	return kv.KeyRange{
+		StartKey: startKey,
+		EndKey:   endKey,
+	}
+}
+
+// convert this key to the smallest key which is larger than the key given.
+// see tikv/src/coprocessor/util.rs for more detail.
+func convertToPrefixNext(key []byte) []byte {
+	if key == nil || len(key) == 0 {
+		key = make([]byte, 1)
+		key[0] = 0
+		return key
+	}
+	for i := len(key) - 1; i >= 0; i -= 1 {
+		if key[i] == 255 {
+			key[i] = 0
+		} else {
+			key[i] += 1
+			return key
+		}
+	}
+	for i := 0; i < len(key); i++ {
+		key[i] = 255
+	}
+	return append(key, 0)
+}
+
+// return whether these two keys are equal.
+func isPrefixNext(key []byte, expected []byte) bool {
+	key = convertToPrefixNext(key)
+	if len(key) != len(expected) {
+		return false
+	}
+	for i := 0; i < len(key); i++ {
+		if key[i] != expected[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func executeAPointGet(data *data, store *TestStore, handler int64) (chunks []tipb.Chunk, err error) {
+	// we would prepare the dagRequest and try build a executor as follows,
+	// the reason that we don't make a tableScanExecutor directly here is
+	// it would be replaced by closureExecutor in the future, so, for compatibility,
+	// we just build a dagReq, and call buildDAGExecutor to build
+	// the corresponding executor.
+	aRange := getTestPointRange(TableId, handler)
+	dagReq := &tipb.DAGRequest{
```

Comment:
If we extract the DAG builder, we can use it for other tests.

Reply:
Yes, but I don't know what it is like for extracting DAG builder, for example, what parameters does it needs, so it might be better that do that when we needs more test using DAG builder, and I can figure out what it likes to extract dag builder. What do you think?